### PR TITLE
refactor: Assign malloc return to a local variable first.

### PR DESCRIPTION
**Pull request by iphydf**
See https://github.com/TokTok/hs-tokstyle/pull/237.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)

### CODE REVIEW

File: toxcore/list.c
Comment (Green-Sky) [inline toxcore/list.c:126]: why reorder?

Comment (iphydf) [reply toxcore/list.c:126]: It's a tokstyle rule we used to apply but then it got lost because we moved to mem.

Comment (Green-Sky) [reply toxcore/list.c:126]: hm, because everywhere else, that I could see from the small code review snippets in gh, it is `sizeof()*N`

Comment (iphydf) [reply toxcore/list.c:126]: "Everywhere else" was 3 locations in the code, which I've fixed now. Let me know if you find any others.

**Review by nurupo — CHANGES_REQUESTED** [review]
> Reviewed 23 of 29 files at r1, 2 of 3 files at r2, 1 of 2 files at r3, 5 of 5 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @Green-Sky and @iphydf)

___
*[`toxcore/group_announce.c` line 371 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfRPA87htbpRUo5ini:-NnfRPA95KVlNPyR2ZqT:b7b6spd) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_announce.c#L371)):*
> ```C
> 
>         announces = new_announces;
>     }
> ```

The original code seemed perfectly fine. `announces` is a local variable and assigning `nullptr` to it as the result of `calloc()` and the doing `return` doesn't result in any side-effects.
I understand that adding additional analysis on local variables of whether assigning `nullptr` and returning would cause any side-effect outside the function scope might be asking too much out of cimple, but thought I would at least point this out.
You can just ack this comment / mark it as resolved.
___
*[`toxcore/group_chats.c` line 4058 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfSM0z8KfofG41Hwdv:-NnfSM0z8KfofG41Hwdw:b6xucbl) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_chats.c#L4058)):*
> ```C
> 
>         memcpy(tmp, chat->shared_state.password, oldlen);
>         oldpasswd = tmp;
> ```

The original `oldpasswd` name is better than `tmp`, can we do something about it?
Could you rename it to `tmp_oldpasswd`?

Also, the original code seemed perfectly fine. `oldpasswd` is a local variable and assigning `nullptr` to it as the result of `malloc()` and the doing `return` doesn't result in any side-effects.
I understand that adding additional analysis on local variables of whether assigning `nullptr` and returning would cause any side-effect outside the function scope might be asking too much out of cimple, but thought I would at least point this out.
___
*[`toxcore/group_connection.c` line 106 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfU2AK0nahvHSDxbjO:-NnfU2AK0nahvHSDxbjP:bz3aeim) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_connection.c#L106)):*
> ```C
> 
>         if (entry_data == nullptr) {
>             array_entry->data = nullptr;  // TODO(iphydf): Is this necessary?
> ```

@JFreegman can you chime in on this one? Is this assignment necessary? Do we also want to set `array_entry->data_length = 0` along with `array_entry->data = nullptr`, or is it fine to keep the non-zero length?

@iphydf's change reproduces what the original code did, but I wonder if the  original behavior of having `array_entry->data = nullptr` and non-zero `array_entry->data_length` is intentional, as that smells like it could result in some logic bug somewhere else. (For example, if some code somewhere else assumes that a non-zero length implies the data is non-null and thus doesn't do a check for the null.)
___
*[`toxcore/mono_time.c` line 122 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-Nnf_Oin0TOAynjY4cLM:-Nnf_Oio3G1_EM6THCiS:bliyatc) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/mono_time.c#L122)):*
> ```C
> 
> #ifndef ESP_PLATFORM
>     pthread_rwlock_t *mutex = (pthread_rwlock_t *)mem_alloc(mem, sizeof(pthread_rwlock_t));
> ```

Could you change it to `lock`, similar to what the original name was? No need to rename it to `mutex`.
(It's also incorrect to call a rwlock a mutex, as rwlock allows multiple readers or one writer, while mutex doesn't differentiate between a reader and writer and allows only one to access it at a time.)

**Review by JFreegman — COMMENTED** [review]
> Reviewed 1 of 29 files at r1, 2 of 3 files at r2, 1 of 2 files at r3, 3 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @Green-Sky and @iphydf)

___
*[`toxcore/group_chats.c` line 4051 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfkY0J2pZT9zoUKY_Y:-NnfkY0J2pZT9zoUKY_Z:bah87tu) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_chats.c#L4051)):*
> ```C
> 
>     if (oldlen > 0) {
>         uint8_t *tmp = (uint8_t *)malloc(oldlen);
> ```

What does this do? `oldpasswd` is a local variable. Maybe I'm misunderstanding the point of this new tokstyle rule.

(I see nurupo beat me to it)
___
*[`toxcore/group_connection.c` line 106 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfU2AK0nahvHSDxbjO:-NnfmHGGArTDM6hmXUwC:b6nox4u) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_connection.c#L106)):*
<details><summary><i>Previously, nurupo wrote…</i></summary><blockquote>

@JFreegman can you chime in on this one? Is this assignment necessary? Do we also want to set `array_entry->data_length = 0` along with `array_entry->data = nullptr`, or is it fine to keep the non-zero length?

@iphydf's change reproduces what the original code did, but I wonder if the  original behavior of having `array_entry->data = nullptr` and non-zero `array_entry->data_length` is intentional, as that smells like it could result in some logic bug somewhere else. (For example, if some code somewhere else assumes that a non-zero length implies the data is non-null and thus doesn't do a check for the null.)
</blockquote></details>

The way we check if an array entry is empty is by checking if the `time_added` field is 0 via the function `array_entry_is_empty()`. So in other words yes it's intentional, and yes it could lead to bugs in the future if someone doesn't use the correct method of checking if the entry is empty. However, this function should only ever be called with an empty array entry, so it shouldn't be possible for the length field to be non-zero if that malloc fails. We could add a redundant empty check at the start of the function to future-proof its intended behaviour, or we could clear the array on failure via `clear_array_entry()`. Or we could leave it as-is and hope that future reviewers catch any bugs.
___
*[`toxcore/group_moderation.c` line 65 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-Nnfr4y7DijlLoWPlAl1:-Nnfr4y7DijlLoWPlAl2:b3o2upy) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_moderation.c#L65)):*
> ```C
> 
>         if (entry == nullptr) {
>             tmp_list[i] = nullptr;  // TODO(iphydf): Is this needed?
> ```

I don't think so since it's already null.

**Review by iphydf — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @Green-Sky, @JFreegman, and @nurupo)

___
*[`toxcore/group_announce.c` line 371 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfRPA87htbpRUo5ini:-Nni2ivA0mCYEBhLxd53:b-bmhfq9) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_announce.c#L371)):*
<details><summary><i>Previously, nurupo wrote…</i></summary><blockquote>

The original code seemed perfectly fine. `announces` is a local variable and assigning `nullptr` to it as the result of `calloc()` and the doing `return` doesn't result in any side-effects.
I understand that adding additional analysis on local variables of whether assigning `nullptr` and returning would cause any side-effect outside the function scope might be asking too much out of cimple, but thought I would at least point this out.
You can just ack this comment / mark it as resolved.
</blockquote></details>

Ok, to make it more obvious, I've factored out the creation-when-null.
___
*[`toxcore/group_chats.c` line 4051 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfkY0J2pZT9zoUKY_Y:-Nni4hK75aq56b58bQ0J:b-pftf0q) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_chats.c#L4051)):*
<details><summary><i>Previously, JFreegman wrote…</i></summary><blockquote>

What does this do? `oldpasswd` is a local variable. Maybe I'm misunderstanding the point of this new tokstyle rule.

(I see nurupo beat me to it)
</blockquote></details>

Ok, reverted.
___
*[`toxcore/group_chats.c` line 4058 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfSM0z8KfofG41Hwdv:-Nni4lU969CRUIq6A-YD:bp3aow6) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_chats.c#L4058)):*
<details><summary><i>Previously, nurupo wrote…</i></summary><blockquote>

The original `oldpasswd` name is better than `tmp`, can we do something about it?
Could you rename it to `tmp_oldpasswd`?

Also, the original code seemed perfectly fine. `oldpasswd` is a local variable and assigning `nullptr` to it as the result of `malloc()` and the doing `return` doesn't result in any side-effects.
I understand that adding additional analysis on local variables of whether assigning `nullptr` and returning would cause any side-effect outside the function scope might be asking too much out of cimple, but thought I would at least point this out.
</blockquote></details>

Reverted.
___
*[`toxcore/group_connection.c` line 106 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfU2AK0nahvHSDxbjO:-Nni4o-p28z2kKTLWCzb:bdnw1a6) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_connection.c#L106)):*
<details><summary><i>Previously, JFreegman wrote…</i></summary><blockquote>

The way we check if an array entry is empty is by checking if the `time_added` field is 0 via the function `array_entry_is_empty()`. So in other words yes it's intentional, and yes it could lead to bugs in the future if someone doesn't use the correct method of checking if the entry is empty. However, this function should only ever be called with an empty array entry, so it shouldn't be possible for the length field to be non-zero if that malloc fails. We could add a redundant empty check at the start of the function to future-proof its intended behaviour, or we could clear the array on failure via `clear_array_entry()`. Or we could leave it as-is and hope that future reviewers catch any bugs.
</blockquote></details>

"hope that reviewers catch bugs" is not a strategy for secure software. Can you make a PR after this one to resolve the TODO and make things safer?
___
*[`toxcore/group_moderation.c` line 65 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-Nnfr4y7DijlLoWPlAl1:-Nni5Bum0i2yZfTnLuId:b-896fix) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_moderation.c#L65)):*
<details><summary><i>Previously, JFreegman wrote…</i></summary><blockquote>

I don't think so since it's already null.
</blockquote></details>

Done.
___
*[`toxcore/mono_time.c` line 122 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-Nnf_Oin0TOAynjY4cLM:-Nni5Gh19xgnHeqgbxIq:b-896fix) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/mono_time.c#L122)):*
<details><summary><i>Previously, nurupo wrote…</i></summary><blockquote>

Could you change it to `lock`, similar to what the original name was? No need to rename it to `mutex`.
(It's also incorrect to call a rwlock a mutex, as rwlock allows multiple readers or one writer, while mutex doesn't differentiate between a reader and writer and allows only one to access it at a time.)
</blockquote></details>

Done.

**Review by JFreegman — COMMENTED** [review]
> Reviewed 21 of 29 files at r1, 1 of 5 files at r4, 1 of 2 files at r5, 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @Green-Sky, @iphydf, and @nurupo)

___
*[`toxcore/net_crypto.c` line 2032 at r6](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnjM9plBS6ycNJjMYKV:-NnjM9plBS6ycNJjMYKW:btzinxv) ([raw file](https://github.com/TokTok/c-toxcore/blob/abdfe496c47a733116870088e44a3d79212a2058/toxcore/net_crypto.c#L2032)):*
> ```C
>     }
> 
>     New_Connection n_c;
> ```

Should `n_c` be initialized to zero? it's not clear that the lines below won't be using the uninitialized fields.

**Review by nurupo — APPROVED** [review]
> Reviewed 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 1 of 2 approvals obtained (waiting on @Green-Sky, @iphydf, and @JFreegman)

___
*[`toxcore/group_connection.c` line 106 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfU2AK0nahvHSDxbjO:-NnkWWTJ8TrzGkYczTR9:b-9emzal) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_connection.c#L106)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

"hope that reviewers catch bugs" is not a strategy for secure software. Can you make a PR after this one to resolve the TODO and make things safer?
</blockquote></details>

@JFreegman I expected you to reply with something definitive and actionable like "let's do X", but instead got a hand-wavy multiple-choice "we could do X, Y, or Z" etc. so which one is it? :D You are the codeowner here, just pick whatever you think is the best and ask @iphydf to correct it!
Agree with @iphydf on having @JFreegman open a PR afterwards addressing this, this is a bit too much of communication/coordination friction and @JFreegman can do it on his own without involving @iphydf. Also this PR doesn't change the behavior, so accepting this PR without @JFreegman's changes is acceptable.

@JFreegman You say that "we check if an array entry is empty is by checking if the `time_added` field is 0 via the function `array_entry_is_empty()`", but while I see the entry being set to empty by setting `data` to `nullptr`, I don't see its `time_added` correspondingly being set to 0 in `create_array_entry()` function. Why not set it all in one place? Also, the function takes `mono_time` argument which is unused for some reason? Anyway, open a new PR :P 
___
*[`toxcore/group_chats.c` line 4058 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfSM0z8KfofG41Hwdv:-Nnk_gcI7x7L0ClRyg5_:b-qlgjzc) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_chats.c#L4058)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

Reverted.
</blockquote></details>

Ack the revert.

(Just to make it clear, what I meant is: if possible, I would like for you to improve cimple to have it recognize that there are no issues with what the code does in this particular and similar cases and not trip on them. However if that's a too big of an ask (hard to implement, or not a priority, or you just don't want to) then this change is passable; while I'm not too happy with it because it's redundant in this case, I understand why the cimple check is done and it does make things better overall, just wish cimple was smarter about it.)

**Review by JFreegman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 0 of 1 approvals obtained (waiting on @Green-Sky, @iphydf, and @nurupo)

___
*[`toxcore/group_connection.c` line 106 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfU2AK0nahvHSDxbjO:-NnkgWx2DEXWbadHXWlz:b-mkk1n1) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_connection.c#L106)):*
<details><summary><i>Previously, nurupo wrote…</i></summary><blockquote>

@JFreegman I expected you to reply with something definitive and actionable like "let's do X", but instead got a hand-wavy multiple-choice "we could do X, Y, or Z" etc. so which one is it? :D You are the codeowner here, just pick whatever you think is the best and ask @iphydf to correct it!
Agree with @iphydf on having @JFreegman open a PR afterwards addressing this, this is a bit too much of communication/coordination friction and @JFreegman can do it on his own without involving @iphydf. Also this PR doesn't change the behavior, so accepting this PR without @JFreegman's changes is acceptable.

@JFreegman You say that "we check if an array entry is empty is by checking if the `time_added` field is 0 via the function `array_entry_is_empty()`", but while I see the entry being set to empty by setting `data` to `nullptr`, I don't see its `time_added` correspondingly being set to 0 in `create_array_entry()` function. Why not set it all in one place? Also, the function takes `mono_time` argument which is unused for some reason? Anyway, open a new PR :P
</blockquote></details>

A null `data` field with a 0 `length` field doesn't indicate an empty entry, as there can be packets with no data that are just used as an indication of something. That's why we allow packets with a null data pointer, and use `time_added` instead to do the empty check. @iphydf On second thought, the assigning data to null is not actually necessary. There's no case where it isn't null already. You could call `clear_array_entry()` if you really want to make sure the entry gets cleared, but it's unnecessary.

As for a definitive answer, I'll think about it some more and open a pull request after this one is merged (though I'd dispute the idea that I'm the code owner). To be clear, the current code is correct as far as I can tell. It's just not future-bug-proof.

**Review by JFreegman — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @Green-Sky, @iphydf, and @nurupo)

___
*[`toxcore/group_connection.c` line 97 at r6](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnkpDo_EK_R8NDEIw5B:-NnkpDo_EK_R8NDEIw5C:b-xs4847) ([raw file](https://github.com/TokTok/c-toxcore/blob/abdfe496c47a733116870088e44a3d79212a2058/toxcore/group_connection.c#L97)):*
> ```C
>     if (length == 0) {
>         array_entry->data = nullptr;
>         array_entry->data_length = 0;
> ```

<strike>These two fields are being assigned twice if this clause is triggered. The second time below is with a `memcpy(ptr, nullptr, 0)`. Can't recall if memcpy with a src=null and length=0 is UB or not (I think it might be), but it's not exactly an improvement either way. The memcpy should only be called if there's something there to copy.</strike>

Edit: Nevermind, just misread the end bracer here.

**Comment by JFreegman** [comment]
*[`toxcore/group_connection.c` line 106 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfU2AK0nahvHSDxbjO:-Nnkq_6B5LHb3zVOZvLF:b-tqvtqu) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_connection.c#L106)):*
<details><summary><i>Previously, JFreegman wrote…</i></summary><blockquote>

A null `data` field with a 0 `length` field doesn't indicate an empty entry, as there can be packets with no data that are just used as an indication of something. That's why we allow packets with a null data pointer, and use `time_added` instead to do the empty check. @iphydf On second thought, the assigning data to null is not actually necessary. There's no case where it isn't null already. You could call `clear_array_entry()` if you really want to make sure the entry gets cleared, but it's unnecessary.

As for a definitive answer, I'll think about it some more and open a pull request after this one is merged (though I'd dispute the idea that I'm the code owner). To be clear, the current code is correct as far as I can tell. It's just not future-bug-proof.
</blockquote></details>

Actually the current code is possibly not correct, blocking until that's addressed.

**Review by JFreegman — APPROVED** [review]
> Reviewed 1 of 5 files at r4, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 0 of 1 approvals obtained (waiting on @Green-Sky, @iphydf, and @nurupo)

**Review by nurupo — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 1 of 2 approvals obtained (waiting on @Green-Sky, @iphydf, and @JFreegman)

___
*[`toxcore/group_connection.c` line 106 at r5](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnfU2AK0nahvHSDxbjO:-Nnl07XN1UqfwwYVmNy5:b-jevgj3) ([raw file](https://github.com/TokTok/c-toxcore/blob/b26351db81b967a9ee186413145d0b6e2a5f504c/toxcore/group_connection.c#L106)):*
>Also, the function takes `mono_time` argument which is unused for some reason?

Oh, my bad, it is used, had to expand the code in Reviewable, thought the function ended earlier.

**Review by Green-Sky — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r8.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: 1 change requests, 1 of 2 approvals obtained (waiting on @iphydf and @JFreegman)

___
*[`toxcore/net_crypto.c` line 1889 at r9](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-Nnn8HeyE5E4dwIkfqyJ:-Nnn8HeyE5E4dwIkfqyK:b-e1gepz) ([raw file](https://github.com/TokTok/c-toxcore/blob/0d45e2038011a6af05e2325cab859dc2184fa8ba/toxcore/net_crypto.c#L1889)):*
> ```C
>         c->crypto_connections[id].last_packets_left_rem = 0;
>         c->crypto_connections[id].packet_send_rate_requested = 0;
>         c->crypto_connections[id].last_packets_left_requested_rem = 0;
> ```

please use float/double literals, so the meaning is obvious.
(`0.f` and `0.` or `0.0f` and `0.0`)

**Review by iphydf — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530)* status: <span>:shipit:</span> complete! 1 change requests, 1 of 1 approvals obtained (waiting on @Green-Sky and @JFreegman)

___
*[`toxcore/group_connection.c` line 97 at r6](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnkpDo_EK_R8NDEIw5B:-Nnn9Z6j9Nj_DemAvlws:bf8ixqn) ([raw file](https://github.com/TokTok/c-toxcore/blob/abdfe496c47a733116870088e44a3d79212a2058/toxcore/group_connection.c#L97)):*
<details><summary><i>Previously, JFreegman wrote…</i></summary><blockquote>

These two fields are being assigned twice if this clause is triggered. The second time below is with a `memcpy(ptr, nullptr, 0)`. Can't recall if memcpy with a src=null and length=0 is UB or not (I think it might be), but it's not exactly an improvement either way. The memcpy should only be called if there's something there to copy.
</blockquote></details>

No memcpy with nullptr happening anywhere.
___
*[`toxcore/net_crypto.c` line 2032 at r6](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-NnjM9plBS6ycNJjMYKV:-Nnn9W6U2UKmgIq9Z5x7:b-896fix) ([raw file](https://github.com/TokTok/c-toxcore/blob/abdfe496c47a733116870088e44a3d79212a2058/toxcore/net_crypto.c#L2032)):*
<details><summary><i>Previously, JFreegman wrote…</i></summary><blockquote>

Should `n_c` be initialized to zero? it's not clear that the lines below won't be using the uninitialized fields.
</blockquote></details>

Done.
___
*[`toxcore/net_crypto.c` line 1889 at r9](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-Nnn8HeyE5E4dwIkfqyJ:-Nnn9Jqr3dZ-jyhaJoEd:b-896fix) ([raw file](https://github.com/TokTok/c-toxcore/blob/0d45e2038011a6af05e2325cab859dc2184fa8ba/toxcore/net_crypto.c#L1889)):*
<details><summary><i>Previously, Green-Sky (Erik Scholz) wrote…</i></summary><blockquote>

please use float/double literals, so the meaning is obvious.
(`0.f` and `0.` or `0.0f` and `0.0`)
</blockquote></details>

Done.

**Review by Green-Sky — APPROVED** [review]
:lgtm_strong:

> Reviewed 3 of 3 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2530#-:-NnnK5XiDTZ7SlTFtGEN:b-jbsf8z)* status: <span>:shipit:</span> complete! 1 of 1 approvals obtained (waiting on @JFreegman)

**Pull request merged**
